app/facades/geocoder_facade.rb

```diff
@@ -2,9 +2,29 @@ class GeocoderFacade
   def self.get_coordinates(address, city, state)
     location = "#{address} #{city},#{state}"
     json_response = GeocoderService.get_coordinates(location)
-    lat_long = {
-      lat: json_response[:results][0][:locations][0][:latLng][:lat],
-      lng: json_response[:results][0][:locations][0][:latLng][:lng]
-    }
+    if GeocoderFacade.validate_location?(address, city, state, json_response)
+      lat_long = {
+        lat: json_response[:results][0][:locations][0][:latLng][:lat],
+        lng: json_response[:results][0][:locations][0][:latLng][:lng]
+      }
+    else
+      {}
+    end
+  end
+
+  def self.validate_location?(address, city, state, json_response)
+    street = json_response[:results][0][:locations][0][:street]
+    response_city = json_response[:results][0][:locations][0][:adminArea5]
+    response_state = json_response[:results][0][:locations][0][:adminArea3]
+    if street.include?('[') || street == ''
```

Comment:
I would clean up this logic ladder somehow. 

you might pull each of those checks into a single method, write a test for it and then do one check instead of a bunch, since the first three all return false. 

so personally I'd do something like this...


```
def self.validate_location?(address, city, state, json_response)
    street = json_response[:results][0][:locations][0][:street]
    response_city = json_response[:results][0][:locations][0][:adminArea5]
    response_state = json_response[:results][0][:locations][0][:adminArea3]

    valid_street?(street) && valid?(response_city, city) && valid(response_state, state)
end

private 

def valid?(expected, actual)
  expected == actual
end

def valid_street?(street)
 !street.include?('[') || street != ''
end
```

Reply:
this is a combination of both of these ideas here: https://refactoring.guru/extract-variable 
https://refactoring.guru/replace-temp-with-query

Reply:
Noted! Thank you!